cmd/service_linux.go

```diff
@@ -0,0 +1,335 @@
+		}
+
+		serviceFile, err := newServiceFile(mountPath, configPath, serviceUser)
+		if err != nil {
+			return fmt.Errorf("error when attempting to create service file: [%s]", err.Error())
+		}
+
+		// run systemctl daemon-reload
+		systemctlDaemonReloadCmd := exec.Command("sudo", "systemctl", "daemon-reload")
+		err = systemctlDaemonReloadCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+
+		// Enable the service to start at system boot
+		systemctlEnableCmd := exec.Command("sudo", "systemctl", "enable", serviceFile)
+		err = systemctlEnableCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+		return nil
+	},
+}
+
+var serviceName string
+var uninstallCmd = &cobra.Command{
+	Use:               "uninstall",
+	Short:             "Uninstall the startup process for Cloudfuse. Requires running as admin.",
+	Long:              "Uninstall the startup process for Cloudfuse. Requires running as admin.",
+	SuggestFor:        []string{"uninst", "uninstal"},
+	Example:           "cloudfuse service uninstall --mount-path=<path/to/mount/path>",
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		// get absolute path of provided relative mount path
+		if strings.Contains(serviceName, ".") || strings.Contains(serviceName, "..") {
+			dir, err := os.Getwd()
+			if err != nil {
+				return fmt.Errorf("error: [%s]", err.Error())
+			}
+			mountPath = common.JoinUnixFilepath(dir, serviceName)
+
+		}
+
+		// get service file name and service file path
+		folderList := strings.Split(serviceName, "/")
+		serviceName = folderList[len(folderList)-1] + ".service"
+		servicePath := "/etc/systemd/system/" + serviceName
+
+		// delete service file
+		removeFileCmd := exec.Command("sudo", "rm", servicePath)
+		err := removeFileCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to delete "+serviceName+" file from /etc/systemd/system due to following error: [%s]", err.Error())
+		}
+
+		// reload daemon
+		systemctlDaemonReloadCmd := exec.Command("sudo", "systemctl", "daemon-reload")
+		err = systemctlDaemonReloadCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+		return nil
+	},
+}
+
+//--------------- command section ends
+
+func newServiceFile(mountPath string, configPath string, serviceUser string) (string, error) {
+	serviceTemplate := ` [Unit]
+	Description=Cloudfuse is an open source project developed to provide a virtual filesystem backed by S3 or Azure storage.
+	After=network-online.target
+	Requires=network-online.target
+
+	[Service]
+	# User service will run as.
+	User={{.ServiceUser}}
+	# Path to the location Cloudfuse will mount to. Note this folder must currently exist.
+	Environment=MountingPoint={{.MountPath}}
```

Comment:
You could simplify this environment section. I only did it this way to make it easy for a user to edit. But if you write it then you could just put the mountPath in the mount and unmount commands directly.